### PR TITLE
docs: beef up the `ProjectBuilder`'s method docstrings

### PR DESCRIPTION
**Pull request by layday**

### CODE REVIEW

**Review by gaborbernat — APPROVED** [review]

File: src/build/__init__.py [hunk 1]
```diff
@@ -169,6 +169,9 @@ def __init__(self, srcdir, config_settings=None, python_executable=sys.executabl
 
     @property
     def python_executable(self):  # type: () -> Union[bytes, Text]
+        """
+        The Python executable used by :mod:`pep517` to invoke the backend.
```
Comment (FFY00) [inline src/build/__init__.py:173]: Using `pep517` is an implementation detail, please remove the reference.

**Review by FFY00 — CHANGES_REQUESTED** [review]

File: src/build/__init__.py (hunk 1 above)
Comment (layday) [reply src/build/__init__.py]: Removed :+1:

Comment (gaborbernat) [reply src/build/__init__.py]: I considered he made the reference here as pep 517 the API, not the package. And in that sense the sentence is right and does not exposes implementation details.

Comment (layday) [reply src/build/__init__.py]: It was meant for the package.  I'm not sure what the 'executable used by the PEP 517 API' would mean in this context.

Comment (gaborbernat) [reply src/build/__init__.py]: Well the python executable hosting the build backend (as opposed the build frontend - current process), https://www.python.org/dev/peps/pep-0517/#build-backend-interface, what else could it?

Comment (FFY00) [reply src/build/__init__.py]: Yeah, `:mod:` references a module.

**Review by FFY00 — APPROVED** [review]

File: src/build/__init__.py (hunk 1 above)
Comment (layday) [reply src/build/__init__.py]: @gaborbernat I mean that it'd have been confusing to introduce another concept (the 'API') in addition to the frontend and backend - it'd have been somewhat of a tautology.  If we wanted to throw PEP 517 in there, I'd have rephrased it as:

> The Python executable used to invoke the PEP 517 backend.

But I think it goes without saying that the backend conforms to PEP 517 - build is a PEP 517 frontend.

**Pull request merged**